index.js

```diff
@@ -54,7 +54,20 @@ async function main() {
     const restored = await cache.restoreCache(cachePaths, cacheKey);
     const ret = await exec.exec('pre-commit', args, {ignoreReturnCode: push});
     if (!restored) {
-        await cache.saveCache(cachePaths, cacheKey);
+        try {
+            await cache.saveCache(cachePaths, cacheKey);
+        } catch (e) {
+            core.warning(
+                [
+                    "There was an error saving the pre-commit environments to cache:\n\n",
+                    `${e.message}\n\n`,
+                    "This only has performance implications and won't change the result of your pre-commit tests.\n",
+                    "If this problem persists on your default branch, you can try to fix it by editing your '.pre-commit-config.yaml'. ",
+                    "I.e. run 'pre-commit autoupdate' or simply add a blank line. ",
+                    "This will result in a different hash value and thus a different cache target.",
+                ].join("")
```

Comment:
I think you can  use template strings as a multi line literal?

Reply:
Yeah, but than you have an indentation issue, same as with python multiline string.
In python we have `textwrap.dedent` in the stdlib for that, but I don't know of something similar in node, without add a new dependency.
And manually breaking the indentation, by dedenting the lines inside the string to the very left, IMHO looks horrible.

Reply:
When using a multiline string literal the only option to remove the indentation w/o adding an additional dependency I found would be to replace double spaces with an empty string `.replace(/  +/g, '')`. 
Would you prefer the format as it is now (only one linebreak after the first line) or rather formatted like the source?

Reply:
Went with the source format, since replacing the double spaces and using line continuation would remove spaces between sentences.